File: src/Algorithm/Random/Random.py
import time
from time import process_time
from src.utils.help import randomSolution, routeLength
from sys import maxsize


def RandomSolution(tsp, town, n, s):
    t_end = time.time() + s
    bestCost = maxsize
    bestPath = []
    t1_start = process_time()
    while time.time() < t_end:
        path = randomSolution(n, town)
        cost = routeLength(tsp, path)
        if bestCost > cost:
            bestCost = cost
            bestPath = path.copy()

    t1_stop = process_time()
    return bestPath, bestCost, (t1_stop - t1_start)

File: src/AlgosC/C/Random/Random.c
#include <limits.h>
#include <stdio.h>
#include <stdlib.h>
#include "Random.h"

void randomPath(int** tsp,int town,int n,int* path,int permutations){
    srand(time(NULL));
    int *bestPath= malloc(sizeof(int)*n);
    int bestCost=INT_MAX,cost;

    for (int i = 0; i < permutations; ++i) {
        randomSolution(n,town,path);
        cost= routeLength(tsp,n,path);
        if(bestCost>cost){
            copyArray(bestPath,path,n);
            bestCost=cost;
        }
    }
    copyArray(path,bestPath,n);
    free(bestPath);
}

